cw/cw305/simple_capture_traces.py

```diff
@@ -145,18 +146,77 @@ def capture_aes(ot, ktp):
         yield ret
 
 
+def capture_aes_fvsr_key(ot):
+    """A generator for capturing AES traces for fixed vs random key test.
+    The data collection method is based on the derived test requirements (DTR) for TVLA:
+    https://www.rambus.com/wp-content/uploads/2015/08/TVLA-DTR-with-AES.pdf
+    The measurements are taken by using either fixed or randomly selected key.
+    In order to simplify the analysis, the first sample has to use fixed key.
+    The initial key and plaintext values as well as the derivation methods are as specified in the
+    DTR.
+
+    Args:
+      ot: Initialized OpenTitan target.
+    """
+    key_generation = bytearray([0x12, 0x34, 0x56, 0x78, 0x9A, 0xBC, 0xDE, 0xF1,
+                                0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xE0, 0xF0])
+    cipher_gen = AES.new(bytes(key_generation), AES.MODE_ECB)
+    text_fixed = bytearray([0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA,
+                            0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA])
+    text_random = bytearray([0xCC, 0xCC, 0xCC, 0xCC, 0xCC, 0xCC, 0xCC, 0xCC,
+                             0xCC, 0xCC, 0xCC, 0xCC, 0xCC, 0xCC, 0xCC, 0xCC])
+    key_fixed = bytearray([0x81, 0x1E, 0x37, 0x31, 0xB0, 0x12, 0x0A, 0x78,
+                           0x42, 0x78, 0x1E, 0x22, 0xB2, 0x5C, 0xDD, 0xF9])
+    key_random = bytearray([0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53,
+                            0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53, 0x53])
+
+    tqdm.write(f'Fixed key: {binascii.b2a_hex(bytes(key_fixed))}')
+
+    sample_fixed = 1
+    while True:
+        if sample_fixed:
+            text_fixed = bytearray(cipher_gen.encrypt(text_fixed))
+            key, text = key_fixed, text_fixed
+        else:
+            text_random = bytearray(cipher_gen.encrypt(text_random))
+            key_random = bytearray(cipher_gen.encrypt(key_random))
+            key, text = key_random, text_random
+        sample_fixed = random.randint(0, 1)
+
+        cipher = AES.new(bytes(key), AES.MODE_ECB)
```

Comment:
Theoretically, you could just initialize the cipher for the fixed key once (now it's done whenever you use it). But I think it doesn't really matter as for the simple capture we don't care about performance. So this looks good to me.

Reply:
To do this, we need to have two different cipher object. One for the fixed key and only initialize once. And the other one for the random key and initialize cipher every time. Which one should we prefer? Current structure or with the two different cipher object.